### PR TITLE
CORE-1841, CORE-1842: Add handlers for listing and adding addons.

### PR DESCRIPTION
**Pull request by johnworth**
Adds handlers for listing and adding available addons to QMS.

See: 
* https://cyverse.atlassian.net/browse/CORE-1841
* https://cyverse.atlassian.net/browse/CORE-1842

### CODE REVIEW

File: app/addons.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	if reqAddon.ResourceType.Name == "" && reqAddon.ResourceType.Uuid == "" {
+		sendError(ctx, response, errors.New("resource_type.name or resource_type.uuid must be set"))
+		return
+	}
+
+	var lookupRT *db.ResourceType
+
+	if reqAddon.ResourceType.Name != "" && reqAddon.ResourceType.Uuid == "" {
```
Comment (slr71) [inline app/addons.go:66]: It's probably not strictly necessary in this case, but I'm a fan of creating an explicit transaction whenever multiple SQL statements are executed for a single operation, even when only one of those SQL statements is an update. To me, having a transaction present already seems to make it less likely that a bug will be introduced later if another database update has to be added for the same logical operation.

File: app/addons.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	newAddon.ResourceType = *lookupRT
+
+	newID, err := d.AddAddon(ctx, newAddon)
+	if err != nil {
+		sendError(ctx, response, err)
+		return
+	}
+
+	response.Addon = newAddon.ToQMSType()
+	response.Addon.Uuid = newID
```
Comment (slr71) [inline app/addons.go:95]: I like that the response body includes all of the information about the add-on. 👍

**Review by slr71 — APPROVED** [review]
I didn't see anything that would cause a problem. :shipit: whenever you're ready!

File: app/addons.go (hunk 1 above)
Comment (johnworth) [reply app/addons.go:66]: This should be fixed now. Thanks for pointing it out.

**Pull request merged**